locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,440 @@
+body after they are handed over to the request handler is not a vulnerability in
+Node.js itself, since it's the responsibility of the application code to handle
+it correctly.
+
+Ensure that the WebServer handle socket errors properly, for instance, when a
+server is created without a error handling, it will be vulnerable to DoS
+
+```js
+const net = require('net');
+
+const server = net.createServer(function(socket) {
+  // socket.on('error', console.error) // this prevents the server to crash
+  socket.write('Echo server\r\n');
+  socket.pipe(socket);
+});
+
+server.listen(5000, '0.0.0.0');
+```
+
+If a _bad request_ is performed the server could crash.
+
+An example of a DoS attack that is not caused by the request's contents is
+[Slowloris][]. In this attack, HTTP requests are sent slowly and fragmented,
+one fragment at a time. Until the full request is delivered, the server will
+keep resources dedicated to the ongoing request. If enough of these requests
+are sent at the same time, the amount of concurrent connections will soon reach
+its maximum resulting in a denial of service. This is how the attack depends
+not on the request's contents, but on the timing and pattern of the requests
+being sent to the server.
+
+**Mitigations**
+
+* Use a reverse proxy to receive and forward requests to the Node.js application.
+Reverse proxies can provide caching, load balancing, IP blacklisting, etc. which
+reduce the probability of a DoS attack being effective.
+* Correctly configure the server timeouts, so that connections that are idle or
+where requests are arriving too slowly can be dropped. See the different timeouts
+in [`http.Server`][], particularly `headersTimeout`, `requestTimeout`, `timeout`,
+and `keepAliveTimeout`.
+* Limit the number of open sockets per host and in total. See the [http docs][],
+particularly `agent.maxSockets`, `agent.maxTotalSockets`, `agent.maxFreeSockets`
+and `server.maxRequestsPerSocket`.
+
+### DNS Rebinding (CWE-346)
+
+This is an attack that can target Node.js applications being run with the
+debugging inspector enabled using the [--inspect switch][].
+
+Since websites opened in a web browser can make WebSocket and HTTP requests,
+they can target the debugging inspector running locally.
+This is usually prevented by the [same-origin policy][] implemented by modern
+browsers, which forbids scripts from reaching resources from different origins
+(meaning a malicious website cannot read data requested from a local IP address).
+
+However, through DNS rebinding, an attacker can temporarily control the origin
+for their requests so that they seem to originate from a local IP address.
+This is done by controlling both a website and the DNS server used to resolve
+its IP address, see [DNS Rebinding wiki][] for more details.
+
+This is not considered a vulnerability in Node.js itself, unless the request has
+an invalid origin (such as an invalid IP, or any domain other than localhost)
```

Comment:
do we mean the loopback address or localhost? localhost can have a public IP technically (I've been on machines where that is true)

Reply:
The point is to explain that when a machine is affected by a DNS attack and the user tries to load a malicious server, gets redirected and eventually exposes the inspector machine is not considered a Node.js vulnerability per see. Any idea how to make it clear?

Reply:
Maybe this paragraph can be removed?